hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -332,7 +332,7 @@ class HoodieCopyOnWriteIncrementalHadoopFsRelationFactory(override val sqlContex
   extends HoodieCopyOnWriteSnapshotHadoopFsRelationFactory(sqlContext, metaClient, options, schemaSpec, isBootstrap) {
 
   override val mandatoryFields: Seq[String] = Seq(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.COMMIT_TIME_METADATA_FIELD) ++
-    preCombineFieldOpt.map(Seq(_)).getOrElse(Seq())
+    preCombineFieldOpt.map(Seq(_)).getOrElse(Seq()) ++ (if (isBootstrap) Seq.empty else  partitionColumnsToRead.toSeq)
```

Comment:
`(if (isBootstrap) Seq.empty else partitionColumnsToRead.toSeq)` is used many times.  Wrap that into a util method.

Reply:
Or reuse `mandatoryFields`.